pwndbg/gdblib/tui/context.py

```diff
@@ -186,25 +186,29 @@ def _ansi_substr(self, line: str, start_char: int, end_char: int) -> str:
                 colored_end_idx += 1
                 char_count += 1
             colored_idx += 1
-        return (
-            (
-                ansi_escape_before_start
-                + line[colored_start_idx:colored_end_idx]
-                + ansi_escape_after_end
-                # Workaround for display bug in GDB TUI ANSI escape sequence translation.
-                # Only resetting the foreground or background colors results in
-                # generating wrong colors in the TUI.
-                # The workaround is to avoid the sequences that only reset the colors
-                # and replace them with a full reset sequence.
-                # It resets other styling attributes like bold too but it's better
-                # than having wrong colors.
-                # https://github.com/pwndbg/pwndbg/issues/2654
-            )
-            .replace("\x1b[39m", "\x1b[0m")
-            .replace("\x1b[49m", "\x1b[0m")
-            .replace("\x1b[39;49m", "\x1b[0m")
+        output = (
+            ansi_escape_before_start
+            + line[colored_start_idx:colored_end_idx]
+            + ansi_escape_after_end
         )
 
+        # Workaround for display bug in GDB TUI ANSI escape sequence translation.
+        # Only resetting the foreground or background colors results in
+        # generating wrong colors in the TUI.
+        # The workaround is to avoid the sequences that only reset the colors
+        # and replace them with a full reset sequence.
+        # It resets other styling attributes like bold too but it's better
+        # than having wrong colors.
+        # https://github.com/pwndbg/pwndbg/issues/2654
+        gdb_version = tuple(map(int, gdb.VERSION.split(".")[:2]))
+        if gdb_version < (16, 3):
+            return (
+                output.replace("\x1b[39m", "\x1b[0m")
+                .replace("\x1b[49m", "\x1b[0m")
+                .replace("\x1b[39;49m", "\x1b[0m")
+            )
+        return output
```

Comment:
Can we do it like this?:

```py
def _ansi_substr(...):
    ...

if gdb_version < (16, 3):
    ___ansi_substr = _ansi_substr
    _ansi_substr = lambda *a, **kw: ___ansi_substr(*a, **kw).replace("\x1b[39m", "\x1b[0m").replace("\x1b[49m", "\x1b[0m").replace("\x1b[39;49m", "\x1b[0m")
```

or sth like this? This way, we won't do the check/version fetching on newer versions.

Reply:
Sure